pkg/platform/kube/monitoring/test/function_test.go

```diff
@@ -0,0 +1,312 @@
+package test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube"
+	"github.com/nuclio/nuclio/pkg/platform/kube/monitoring"
+	kubetest "github.com/nuclio/nuclio/pkg/platform/kube/test"
+
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FunctionMonitoringTestSuite struct {
+	kubetest.KubeTestSuite
+}
+
+func (suite *FunctionMonitoringTestSuite) SetupSuite() {
+	suite.KubeTestSuite.SetupSuite()
+
+	// decrease blocking interval, to make test run faster
+	monitoring.PostDeploymentMonitoringBlockingInterval = time.Second
+}
+
+func (suite *FunctionMonitoringTestSuite) TestNoRecoveryAfterBuildError() {
+	functionName := "function-build-fail"
+	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
+	getFunctionOptions := &platform.GetFunctionsOptions{
+		Name:      createFunctionOptions.FunctionConfig.Meta.Name,
+		Namespace: createFunctionOptions.FunctionConfig.Meta.Namespace,
+	}
+
+	// fail during build
+	createFunctionOptions.FunctionConfig.Spec.Build.Commands = []string{
+		"exit 1",
+	}
+
+	// wait for at least one monitor + post deployment blocking intervals
+	waitOnMonitoringIntervals := 2*suite.Controller.GetFunctionMonitoringInterval() +
+		monitoring.PostDeploymentMonitoringBlockingInterval
+
+	suite.DeployFunctionExpectErrorAndRedeploy(createFunctionOptions,
+		func(deployResult *platform.CreateFunctionResult) bool {
+
+			// wait for monitoring
```

Comment:
I'm assuming `DeployFunctionExpectErrorAndRedeploy` already does `suite.GetFunctionAndExpectState(getFunctionOptions, functionconfig.FunctionStateError)` before this sleep. right?

Reply:
It does, in other way but yes.

The reason im waiting for monitor is not because _i want it_ to change the state but the opposite. I want to ensure it did not change function state.

Now that im looking at the test, I should change its flow to
1. successfully deploy a function
2. make its build command fail
3. expect function state to remain in error (although it has a running replica)